src/Autofac/Core/Resolving/InstanceLookup.cs

```diff
@@ -102,9 +102,6 @@ public object Execute()
                 _activationScope,
                 resolveParameters);
 
-            if (instance != decoratorTarget)
```

Comment:
Unless I'm missing something specific, you wouldn't actually want to raise the Activating event again for the same registration, but specify the decorated instance. CreateInstance() already raises Activating for the original, so this block that's removed is a secondary occurrence only happening with decorators in play.